frontend/src/components/pages/Dashboard/components/DeleteRecurringModal.tsx

```diff
@@ -0,0 +1,78 @@
+import {
+  Button,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Radio,
+  RadioGroup,
+  Stack,
+  Text,
+} from "@chakra-ui/react";
+import React, { useState } from "react";
+
+import useViewport from "../../../../hooks/useViewport";
+
+interface DeleteRecurringModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  onDelete: (isOneTimeEvent?: boolean) => void;
+}
+const DeleteRecurringModal = ({
+  isOpen,
+  onClose,
+  onDelete,
+}: DeleteRecurringModalProps) => {
+  const { isDesktop } = useViewport();
+  const [deleteScheduleValue, setDeleteScheduleValue] = useState("this");
+
+  return (
+    <>
+      <Modal
+        size={isDesktop ? "2xl" : "xs"}
+        isOpen={isOpen}
+        onClose={onClose}
+        isCentered
+      >
+        <ModalOverlay />
+        <ModalContent py="52px" px="48px">
+          <ModalHeader>
+            <Text textStyle={isDesktop ? "desktopHeader3" : "mobileBodyBold"}>
+              Cancel Recurring Donation
+            </Text>
+          </ModalHeader>
+          <ModalCloseButton />
+          <ModalBody textStyle="mobileBody" color="hubbard.100">
+            <RadioGroup
+              onChange={setDeleteScheduleValue}
+              value={deleteScheduleValue}
+            >
+              <Stack>
+                <Radio size="lg" colorScheme="red" value="this">
+                  This donation
+                </Radio>
+                <Radio size="lg" colorScheme="red" value="all">
+                  This and all following donations
```

Comment:
The way its implemented currently all the donations will be cancelled instead of only `this and all following donations`. Either we should change the wording or edit the backend function so that only those which have a start time greater than the one trying to cancel will be deleted (this should be simple as well)

Reply:
fixed to address the this and all following donations